### DISCUSSION TITLE
How to apply custom tasklist design?

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#1256] [Q&A] [answered]
I would like to add the custom tasklist item design that is showcased on the [Tasklist page](https://facelessuser.github.io/pymdown-extensions/extensions/tasklist/) but adding it as a CSS doesn't actually apply the change it seems.

I use MkDocs and have the `custom_checkbox` option set to true. I also have the CSS code imported into the main CSS file that is set in the mkdocs.yml file.

I would appreciate help with this.

Here's the actual file used.
https://github.com/Andre601/GitHub-Markdown/blob/feature/gfm-tasklist-icons/docs/assets/css/styling/checkbox.css

A preview of the checkbox is here:
https://deploy-preview-8--github-markdown.netlify.app/github/general/#tasks

**Comment by facelessuser** [comment, marked as answer]
If you are using MkDocs Material (which it seems you are) you don't really need to do anything except set `custom_checkbox`. The only additional styling we do in our docs is coloring: https://github.com/facelessuser/pymdown-extensions/blob/main/docs/src/scss/extensions/_tasklist.scss.

The example CSS provided in the documents is only for environments that aren't already doing special styling on the tasklist. If you combine the example CSS on top of Material's custom CSS, you will end up with something not quite like you expect.

**Reply by Andre601** [reply]
My goal here is to have task lists similar to github. And the comment says that the custom CSS example would achieve a similar result (Squares with rounded corners and a green checkmark for selected ones)

**Reply by facelessuser** [reply]
You have to undo the tasklist CSS or modify it to look like GitHub. You can probably replace the icon that Material is using: https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/stylesheets/main/extensions/pymdownx/_tasklist.scss#L28.

Material inlines the SVG, so it is possible you could inline your own in CSS overwriting the `--md-tasklist-icon` and `--md-tasklist-icon--checked` variable.

**Reply by facelessuser** [reply]
Honestly, I don't think GitHub uses custom icons per se, so maybe turning off custom and adding appropriate CSS could make it work.

**Reply by Andre601** [reply]
Yeah. I'm sadly not a big expert with CSS. Will need to take a look at what Github perhaps does to mimic this behaviour more closely.

**Reply by Andre601** [reply]
I quickly tried to override the theme by adding fontawesome icons into `.icons/fontawesome/regular` and `.icons/fontawesome/solid` respectively, followed by creating `assets/stylesheets/main/extensions/pymdownx/_tasklist.scss` where I defined the new Path to load icons from.

But it doesn't load the new SVG icons despite me overriding the theme, double-checking the folder structure and file names and making sure the info is right.

This is the scss file:
https://github.com/Andre601/GitHub-Markdown/blob/feature/gfm-tasklist-icons/theme/assets/stylesheets/main/extensions/pymdownx/_tasklist.scss

Fontawesome icons:
https://github.com/Andre601/GitHub-Markdown/tree/feature/gfm-tasklist-icons/theme/.icons/fontawesome

**Reply by facelessuser** [reply]
This is CSS question. Are you finding the SCSS to CSS? I know my SCSS cults use some SASS extension to inline the SVG images and such. All of this is converted to CSS.

Anyways, I'm certain it is possible, but I  would need to find some time to sit down and code up a proper example. Maybe I'll have a chance this weekend.

**Reply by Andre601** [reply]
Managed to get something working:
https://github.com/squidfunk/mkdocs-material/discussions/2361
